membership-attribute-service/app/services/zuora/payment/ZuoraPaymentService.scala

```diff
@@ -172,9 +156,12 @@ class ZuoraPaymentService(zuoraService: ZuoraSoapService, planMap: Map[ProductRa
     } yield paymentMethod
   }
 
-  private def getPaymentMethodByAccount(account: Account): Future[Option[Queries.PaymentMethod]] =
-    (for {
-      paymentId <- OptionT(Future.successful(account.defaultPaymentMethodId))
-      paymentMethod <- zuoraService.getPaymentMethod(paymentId).liftM
-    } yield paymentMethod).run
+  private def getPaymentMethodByAccount(account: Account): Future[Option[Queries.PaymentMethod]] = {
+    val maybeEventualPaymentMethod = account.defaultPaymentMethodId.map(zuoraService.getPaymentMethod)
+    maybeEventualPaymentMethod match {
+      case Some(eventualPaymentMethod) => eventualPaymentMethod.map(Some.apply)
+      case None => Future.successful(None)
+    }
+  }
```

Comment:
this is slightly more redundant, but it saves people having to learn what an OptionT is